### PR TITLE
Advanced 140 Access: Quiz, content change

### PR DESCRIPTION
**Pull request by leogermond**
Port back changes that were made on the fundamental, those two should ultimately share this set of slides. Add back pool-specific access types, add slide on anonymous access

- adv 140: dfn
- adv 140: put back pool-specific, add slide on anonymous
- adv 140: add back accessibility checks second slide
- adv 140: fix adding back pool-specific access types

### CODE REVIEW

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/adv_140_access_types.rst [hunk 1]
```diff
@@ -128,6 +128,105 @@ Access Types and Primitives
 
          procedure Proc (V : access T); -- Primitive of T
 
+-----------------------
+Anonymous Access Types
+-----------------------
+
+* Can be declared in several places
+
+    + Are **pool-specific**
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_140_access_types.rst:137]: Anonymous are general, not pool-specific

File: courses/fundamentals_of_ada/adv_140_access_types.rst [hunk 2]
```diff
@@ -197,11 +296,38 @@ Introduction to Accessibility Checks
          end Proc;
       end P;
 
-* Access types can access objects at most of the same depth
+* Access types can access objects of the **same or lower** depth
 * The compiler checks it statically
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_140_access_types.rst:300]: "lower" is confusing - you don't mean "deeper", you mean "lower number"

**Review by frank-at-adacore — DISMISSED** [review]

File: courses/fundamentals_of_ada/adv_140_access_types.rst (hunk 1 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_140_access_types.rst:137]: Indeed :facepalm:

File: courses/fundamentals_of_ada/adv_140_access_types.rst (hunk 2 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_140_access_types.rst:300]: Yes, and added an example

**Review by rchoch — APPROVED** [review]

**Pull request merged**